lexer_test.go

```
@@ -0,0 +1,119 @@
+			ok:   false,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			got, ok := tc.lexer.Next()
+			opts := []cmp.Option{
+				cmp.AllowUnexported(SimpleToken{}, EagerLexer{}),
+			}
+			if diff := cmp.Diff(tc.want, got, opts...); diff != "" {
+				t.Fatalf(diff)
```

golangci-lint (1.62.2)

printf: non-constant format string in call to (*testing.common).Fatalf

(govet)